clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -270,6 +275,27 @@ protected RecordSerializer<K> getKeySerializerForRequest() {
         return getKeySerializerWithRetryWithShortInterval();
       } catch (Exception e) {
         whetherStoreInitTriggeredByRequestFail = true;
+        // Kick off an async thread to keep retrying
+        asyncStoreInitThread = new Thread(() -> {
+          while (true) {
+            try {
+              getKeySerializerWithRetryWithShortInterval();
+              whetherStoreInitTriggeredByRequestFail = false;
```

Comment:
Should use atomic/volatile for `whetherStoreInitTriggeredByRequestFail` since write and read happens in different threads now.

Reply:
Good call, will update it to be `volatile`.